infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationDao.kt

```diff
@@ -270,8 +302,13 @@ class PublicationDao(
               from layout.location_track_publication_view lt
                 left join layout.segment_version s on s.alignment_id = lt.alignment_id
                 and s.alignment_version = lt.alignment_version
-              where :publication_status = any(lt.publication_states)
-                and lt.state != 'DELETED'
+              where case
+                when ${if (locationTrackIdsInPublicationUnit == null) "true"
```

Comment:
Tuon iffittelyn voisi ehkä ottaa muuttujaan (esim `const draftTrackIncludedCondition = if...`) sql:n yllä jotta saisi tämän ympäröivän draft/official checkin vähän luettavammaksi

Reply:
Tämä koko iffittelyrakenne menee muutenkin mielestäni turhan monimutkaiseksi, kun `null` on oma spessuarvonsa. Perffioptimointi joo, mutta mikäli tämän haun perffiä ei olla identifioitu ongelmaksi niin itse vaan passaisin tälle funktiolle lähtökohtaisestikin aina non-nullable-listan, jossa sitten vaan on kaikki arvot. Tällöin tämä iffittely säilyisi kertaluokkaa helppolukuisempana.